oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -402,25 +401,22 @@ impl ServerStreamingService<Vec<u8>> for GrpcInvocationHandler {
             let response_iter = handler
                 .inject_grpc_request(grpc_request, &oak_label)
                 .map_err(|()| tonic::Status::new(tonic::Code::Internal, ""))?;
-
-            // First convert the `Iterator<Item = GrpcResponse>` to an
-            // `Iterator<Item = Result<Vec<u8>, tonic::Status>`.
-            let result_iter = response_iter.map(|response| {
-                debug!("Returning gRPC response: {:?}", response);
-                match response.status {
-                    None => Ok(response.rsp_msg),
-                    Some(status) if status.code == rpc::Code::Ok as i32 => Ok(response.rsp_msg),
-                    Some(status) => Err(to_tonic_status(status)),
+            let (tx, rx) = mpsc::unbounded_channel();
+
+            // Spawn a separate thread to handle responses, because the underlying call to
+            // Runtime::wait_on_channels() blocks.
+            std::thread::spawn(move || {
```

Comment:
Is it creating a thread per stream or per request?

Reply:
Both, I think they're 1:1 &ndash; one method invocation corresponds to one incoming request (as we don't support client streaming) which corresponds to one stream of responses.